client/app/bundles/course/assessment/submission/components/ScribingView/ScribingCanvas.jsx

```diff
@@ -728,28 +727,58 @@ export default class ScribingCanvas extends React.Component {
-              newObjects.push(newObj);
+            let newObj = {};
+
+            // Don't wrap single object in group,
+            // in case it's i-text and we want it to be editable at first tap
+            if (this.copiedObjects.length === 1) {
+              const obj = this.copiedObjects[0];
+              if (obj.type === 'i-text') {
+                newObj = this.cloneText(obj);
+              } else {
+                newObj = fabric.util.object.clone(obj);
+              }
+
+              // Shift copied object to the left if there's space
+              this.copyLeft = (this.copyLeft + newObj.getWidth() > this.canvas.getWidth()) ?
+                this.copyLeft : this.copyLeft + 10;
+              newObj.setLeft(this.copyLeft);
+              // Shift copied object down if there's space
+              this.copyTop = (this.copyTop + newObj.getHeight() > this.canvas.getHeight()) ?
+                this.copyTop : this.copyTop + 10;
+              newObj.setTop(this.copyTop);
+
               newObj.setCoords();
               this.canvas.add(newObj);
-              newObj.set('active', true);
-            });
-            const group = new fabric.Group(newObjects, { canvas: this.canvas });
-
-            // Shift copied object to the left if there's space
-            this.copyLeft = (this.copyLeft + group.getWidth() > this.canvas.getWidth()) ?
-              this.copyLeft : this.copyLeft + 10;
-            group.setLeft(this.copyLeft);
-            // Shift copied object down if there's space
-            this.copyTop = (this.copyTop + group.getHeight() > this.canvas.getHeight()) ?
-              this.copyTop : this.copyTop + 10;
-            group.setTop(this.copyTop);
-
-            group.setCoords();
-            this.canvas.setActiveGroup(group);
-            group.saveCoords();
-            this.canvas.renderAll();
+              this.canvas.setActiveObject(newObj);
+              this.canvas.renderAll();
+            } else {  // Cloning a group of objects
+              this.copiedObjects.forEach((obj) => {
+                if (obj.type === 'i-text') {
+                  newObj = this.cloneText(obj);
+                } else {
+                  newObj = fabric.util.object.clone(obj);
+                }
+                newObj.setCoords();
+                this.canvas.add(newObj);
+                newObjects.push(newObj);
+              });
+              const group = new fabric.Group(newObjects, { canvas: this.canvas });
+
+              // Shift copied object to the left if there's space
+              this.copyLeft = (this.copyLeft + group.getWidth() > this.canvas.getWidth()) ?
+                this.copyLeft : this.copyLeft + 10;
+              group.setLeft(this.copyLeft);
+              // Shift copied object down if there's space
+              this.copyTop = (this.copyTop + group.getHeight() > this.canvas.getHeight()) ?
+                this.copyTop : this.copyTop + 10;
+              group.setTop(this.copyTop);
```

Comment:
Maybe can extract the shifting logic out into a method?

Reply:
Ok